### PR TITLE
Update Sapiovore mood boosts from eating humans

### LINKED ISSUE #58764: Sapiovore cancels out Psychopath (and Killer Drive)
### Describe the bug

When I obtain the Sapiovore trait (e. g. from lizard mutation) I lose the Psychopath trait among others, like Killer Drive.

Looking at [this chunk of consumption.cpp](https://github.com/CleverRaven/Cataclysm-DDA/blob/57fd5acf90b6f2930e5d9655b5d3ac8d44643240/src/consumption.cpp#L1183-L1224) especially at the comment in there reading:
```cpp
// Sapiovores don't recognize humans as the same species.
// But let them possibly feel cool about eating sapient stuff - treat like psycho
// However, spiritual sapiovores should still recognize humans as having a soul or special for religious reasons
```
I highly doubt, that this is intended behaviour.

This might be another victim of the mutation rework.
However: IIRC this worked a couple dev builds before, but I have yet to hunt down the change, that broke it.

### Steps to reproduce

1. ~~Start off with the Psychopath trait or~~ Gain Psychopath by mutating through the debug menu.
2. Go for any mutation that 'grants' the Sapiovore trait, e. g. lizard.
3. Mutate until you get the Sapiovore trait by normal means of through the debug menu.

### Expected behavior

You keep the Psychopath trait when gaining the Sapiovore trait.

### Screenshots

grafik

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1766 (21H2)
- Game Version: d4e5c54 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Stadler76**
#### Summary
Content "Update Sapiovore mood boosts from eating humans"

#### Purpose of change

Since you can't be a Sapiovore and a Psychopath at the same time anymore (See issue #58764), mood boosts for eating human meat had to be reorganized. Additionally the ingame message "Meh.  You've eaten worse." has been changed to a more neutral "Mmh.  Tastes like venison." since it makes more sense for them there.

#### Describe the solution

For Sapiovores there are only two cases that remain: Sapiovore with and without Spiritual. So I could clean up the code in consumption.cpp a bit.

The result is:
- Sapiovore and Spiritual combined grant a mood boost of 10.
  Message: "You eat the human flesh, and in doing so, devour their spirit."
  Kudos to @fairyarmadillo for fixing the conditional there, before I changed it again.
  Fun fact: For lizards and raptors you could technically say, that their human god has become [Quetzalcoatl](https://en.wikipedia.org/wiki/Quetzalcoatl).
- For Sapiovore alone the message is omitted, since they don't feel distaste, like a Psychopath.
  A Sapiovore doesn't care much about taste, as long as it's meat.
  A Psychopath on the other hand would eat human meat because he or she had no other choice. If that same Psychopath had a choice between a steak and a cooked creep, he'd rather eat the steak and toss the cooked creep away, hence why I've left the message there for them.

#### Describe alternatives you've considered

~~The only alternative I considered, was omitting the ingame message for sapiovores ("Meh.  You've eaten worse." sounded a bit weird for a human eating creature) and giving no mood boost at all.~~
Scratch that. This 'consideration' actually made it into the PR. See the discussion below.

#### Testing

Did some testing eating cooked creeps with a human with all permutations of Cannibal, Psycho and Spiritual and with a fully mutated lizard with and without Spiritual.

#### Additional context

Table of the current outcomes:
| Cannibal | Psycho | Spiritual | Sapiovore | Morale | Message                                                                      |
|:--------:|:------:|:---------:|:---------:|-------:|------------------------------------------------------------------------------|
|     ✔    |    ✔   |     ✔     |     ❌     |    +25 | "You feast upon the human flesh, and in doing so, devour their spirit."      |
|     ✔    |    ✔   |     ❌     |     ❌     |    +15 | "You feast upon the human flesh."                                            |
|     ✔    |    ❌   |     ✔     |     ❌     |    +15 | "You consume the sacred human flesh."                                        |
|     ❌    |    ❌   |     ✔     |     ✔     |    +10 | "You eat the human flesh, and in doing so, devour their spirit."             |
|     ✔    |    ❌   |     ❌     |     ❌     |    +10 | "You indulge your shameful hunger."                                          |
|     ❌    |    ✔   |     ✔     |     ❌     |     +5 | "You greedily devour the taboo meat."                                        |
|     ❌    |    ✔   |     ❌     |     ❌     |      0 | "Meh.&nbsp;&nbsp;You've eaten worse."                                                  |
|     ❌    |    ❌   |     ❌     |     ✔     |      0 | "Mmh.&nbsp;&nbsp;Tastes like venison."                                                 |
|     ❌    |    ❌   |     ✔     |     ❌     |    -60 | "This is probably going to count against you if there's still an afterlife." |
|     ❌    |    ❌   |     ❌     |     ❌     |    -60 | "You feel horrible for eating a person."                                     |

Closes #58764

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
Why are non spiritual.sapiovores getting a mood bonus from human meat? The point of the mutation is that they don't care.

**Comment by Stadler76** [comment]
> Why are non spiritual sapiovores getting a mood bonus from human meat? The point of the mutation is that they don't care.
> 
> Psychopathy is not a drive to hurt others, it's a state of not caring. Not caring because you have a psychiatric issue plus not caring because you're turning into a lizard shouldn't make you suddenly get off on eating people. This is a nonsensical change.

I state my train of thoughts above, but you're probably right, so BTDT.

**Comment by Stadler76** [comment]
Updated the PR message as per discussion.
For those, who wonder, where I've mentioned issue #49238 there: Originally this PR took that issue into consideration but as of now, that issue and this PR are completely unrelated putting aside the 'change' to spritual sapiovores. I just wanted to clear that up.

**Comment by Stadler76** [comment]
@fairyarmadillo: One final question: Originally `sapiovore && spiritual` gave +15 instead of +25 mood. Shall I change that or do you think, that this PR is good to go as is?

**Comment by fairyarmadillo** [comment]
It was probably there to make up for the fact that sapiovores were losing the cannibal mutation. Since chargen mutations are stuck on now and cannibals can't become sapiovores under any circumstances, I don't see the point.

**Comment by Stadler76** [comment]
> edit: I misread your question! I have no idea where they got those values. I would just make it the same number as spiritual+psychopath as that represents the same thing as spiritual+sapiovore basically.

You mean, reduce it to +5 mood? Might be a bit too low. The comment there says:
```cpp
// Small bonus for violating a taboo.
```

I was thinking more into the direction of reverting this change:
```diff
-        } else if( ( cannibal || sapiovore ) && spiritual ) {
+        } else if( cannibal && spiritual ) {
             add_msg_if_player( m_good, _( "You consume the sacred human flesh." ) );
             // Boosted because you understand the philosophical implications of your actions, and YOU LIKE THEM.
             add_morale( MORALE_CANNIBAL, 15, 200 );
```
But the message doesn't fit very well, mmh.
Maybe change it to: "You eat the human flesh, and in doing so, devour their spirit."?

How about the following:
```cpp
        } else if( sapiovore && spiritual ) {
            add_msg_if_player( m_good, _( "You eat the human flesh, and in doing so, devour their spirit." ) );
            add_morale( MORALE_CANNIBAL, 10, 50 );
```
Aka: Don't treat them on the same level as `cannibal && spiritual` because cannibalism is about eating your own species and humans aren't same species anymore. Would be a compromise between that and 'breaking a taboo' which, too only fits for humans.

**Comment by Stadler76** [comment]
Ok, added the +10 morale chunk and modified the top conditional, so it doesn't handle sapiovores anymore.
I've tested all possible permutations of cannibal / psycho / spiritual / sapiovore and the messages as seen in the current diff to master show up as intended (I think?).
I'd we've done enough nitpicking about this ... and there's always the option of a last minute change from a maintainer.

PS: Ultimately it would be way easier to maintain and test, if that chunk would be jsonified. Especially, since you don't need to run the compiler every time you make a change to it.

File: src/consumption.cpp [hunk 1]
```diff
@@ -1204,27 +1204,32 @@ void Character::modify_morale( item &food, const int nutr )
...
-        } else if( ( psycho || sapiovore ) && spiritual ) {
+        } else if( psycho && spiritual ) {
             add_msg_if_player( _( "You greedily devour the taboo meat." ) );
             // Small bonus for violating a taboo.
             add_morale( MORALE_CANNIBAL, 5, 50 );
-        } else if( psycho || sapiovore ) {
+        } else if( psycho ) {
             add_msg_if_player( _( "Meh.  You've eaten worse." ) );
+        } else if( sapiovore ) {
+            // Omitting the message for sapiovores. They just don't care what meat they eat.
```
Comment (Willenbrink) [inline src/consumption.cpp:1232]: I think some kind of message would still be nice as eating human meat happens rarely and could be acknowledged. Perhaps "Tastes like pork" (or veal/beef, [as serial killers have differing views on the matter](https://www.theguardian.com/science/the-lay-scientist/2010/sep/05/human-meat-taste-cannibal)).

Comment (Willenbrink) [reply src/consumption.cpp]: Yes, sounds good!

**Review by Willenbrink — APPROVED** [review]
Looks good to me.
Just so you're aware, I'm not a maintainer. I also just saw that you already posted in the ready-to-merge channel so I guess I've got nothing more to add. Thanks for contributing :)

**Comment by Stadler76** [comment]
> Looks good to me. Just so you're aware, I'm not a maintainer. I also just saw that you already posted in the ready-to-merge channel so I guess I've got nothing more to add. Thanks for contributing :)

I requested that before I've got an all green on those checks. Actually I expected the opposite to happen at that point.

**Pull request merged**